kots-exporter/kots-exporter.sh

```diff
@@ -359,37 +378,25 @@ output_message(){
     echo "- $path/output/helm-values.yaml"
     echo ""
     echo "-------------------------------------------------------------------------"
-    
-    echo "## Postgres Chart Upgrade Preparation"
-    echo "Upgrading to server CircleCI Server 4.0 includes upgrading the Postgres chart"
-    echo "Before upgrading, we need to prepare your postgres instance."
-    echo "This is only needed if your Postgres Instance is not externalized."
-    echo ""
-    echo "# Collect your postgres user's password and the PVC attached to your postgres instance"
-    echo "export POSTGRESQL_PASSWORD=\$(kubectl get secret --namespace $namespace postgresql -o jsonpath=\"{.data.postgresql-password}\" | base64 --decode)"
-    echo "export POSTGRESQL_PVC=\$(kubectl get pvc --namespace $namespace -l app.kubernetes.io/instance=circleci-server,role=primary -o jsonpath=\"{.items[0].metadata.name}\")"
-    echo "# remove the postgres statefulset without terminating your postgres instance"
-    echo "kubectl delete statefulsets.apps postgresql --namespace $namespace --cascade=orphan"
-    echo "# remove the existing secret containing your postgres password. This will get recreated during upgrade."
-    echo "kubectl delete secret postgresql --namespace $namespace"
-    echo ""
-    echo "-------------------------------------------------------------------------"
-    
+
     echo "## Helm login to cciserver.azurecr.io"
-    echo "export HELM_EXPERIMENTAL_OCI=1"
-    echo "helm registry login cciserver.azurecr.io --username <image-registry-username> --password <image-registry-password>"
+    echo "echo <image-registry-password> | helm registry login cciserver.azurecr.io --username <image-registry-username> --password-stdin"
     echo ""
-    
+
     echo "## Helm Diff (optional)"
     echo "The Helm Diff tool is used to verify that the changes between your current install and the upgrade are expected."
     echo ""
     echo "# diff command"
     echo "helm diff upgrade $slug -n $namespace -f $path/output/helm-values.yaml --show-secrets --context 5 $CHART --version 4.0.0"
     echo ""
     echo "-------------------------------------------------------------------------"
-    
+
     echo "## Helm Upgrade CircleCI Server"
+    if "$POSTGRES_INTERNAL"; then
+    echo "source $path/output/postgres-migration.sh && helm upgrade $slug -n $namespace -f $path/output/helm-values.yaml $CHART --version 4.0.0 --force"
```

Comment:
Why are the commands to delete postgres resources being save to run here?

Reply:
Postgres ENV var which is being saved in `postgres-migration.sh` must be available in `current shell` that is why using `source $path/output/postgres-migration.sh` rather than `./postgres-migration.sh` which creates a `child shell` 

The command will only be echoed if `Postgres` is Internal.